packages/xdl/src/Project.ts

```diff
@@ -2346,6 +2394,9 @@ export async function startAsync(
     await Webpack.restartAsync(projectRoot, options);
     DevSession.startSession(projectRoot, exp, 'web');
     return exp;
+  } else if (getenv.boolish('EXPO_USE_DEV_SERVER', false)) {
+    await startDevServerAsync(projectRoot, options);
+    DevSession.startSession(projectRoot, exp, 'native');
```

Comment:
IIRC we talked about potentially putting this in another file so we could identify and deprecate the legacy code more easily.

Reply:
I'd rather do this as a part of splitting up XDL, I'll start working on it soon.